File: src/main/java/dev/zprestige/fire/module/visual/chams/TotemPopListener.java
package dev.zprestige.fire.module.visual.chams;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.TotemPopEvent;
import net.minecraft.entity.player.EntityPlayer;

public class TotemPopListener extends EventListener<TotemPopEvent, Chams> {

    public TotemPopListener(final Chams chams) {
        super(TotemPopEvent.class, chams);
    }

    @Override
    public void invoke(final Object object) {
        final TotemPopEvent event = (TotemPopEvent) object;
        final EntityPlayer entityPlayer = event.getEntityPlayer();
        if (module.popChams.GetSwitch() && (!entityPlayer.equals(mc.player) || module.popSelf.GetSwitch())) {
            module.addEntity(event.getEntityPlayer());
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/visual/chams/TotemPopListener.kt
package dev.zprestige.fire.module.visual.chams

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.TotemPopEvent

class TotemPopListener(chams: Chams) : EventListener<TotemPopEvent, Chams>(TotemPopEvent::class.java, chams) {

    override fun invoke(e: Any) {
        val event = e as TotemPopEvent
        val entityPlayer = event.entityPlayer
        if (module!!.popChams.GetSwitch() && (entityPlayer != mc.player || module!!.popSelf.GetSwitch())) {
            module!!.addEntity(event.entityPlayer)
        }
    }
}